git/git.go

```diff
@@ -654,6 +654,22 @@ func GitDir() (string, error) {
 	return "", nil
 }
 
+func GitCommonDir() (string, error) {
+	cmd := gitNoLFS("rev-parse", "--git-common-dir")
+	out, err := cmd.Output()
+	if err != nil {
```

Comment:
We have a `IsVersionAtLeast()`, though I suspect that the way it is currently is better, since we'll behave more gracefully in the face of _any_ error by falling back to `GitDir()`.

Reply:
It's funny you should mention that, because I think this is the cause of the failures with older Git.  I just noticed this:

```console
$ git rev-parse --random; echo $?
--random
0
```

Granted, this is on Git 2.19, but assuming that the behavior hasn't changed, that's definitely not going to provide a helpful response on older Git.